Assets/BossRoom/Scripts/Shared/Game/State/GameStateBehaviour.cs

```diff
@@ -53,6 +55,22 @@ public virtual bool Persists
         /// </summary>
         private static GameObject s_ActiveStateGO;
 
+        protected DIScope m_Scope;
+
+        [SerializeField]
+        GameObject[] m_GameObjectsThatWillBeInjectedAutomatically;
+
+        protected virtual void Awake()
+        {
+            InitializeScope();
+            m_Scope.FinalizeScopeConstruction();
+            foreach (var autoInjectedGameObject in m_GameObjectsThatWillBeInjectedAutomatically)
+            {
+                m_Scope.InjectIn(autoInjectedGameObject);
+            }
+            DIScope.RootScope.InjectIn(this);
```

Comment:
I think that this line should be the first one in the method, no? Because the InitializeScope could be referring to wahtever dependencies that are in the child class while it's running InitializeScope function